pynicotine/gtkgui/search.py

```diff
@@ -789,6 +792,46 @@ def set_filters(self, stored_filters):
 
         self.on_refilter()
 
+    def process_folder_results(self, folder_results):
+
+        if self.found_include_filter and not self.found_exclude_filter:
+            for row in folder_results:
+                self.add_row_to_model(row)
+
+        folder_results.clear()
+        self.found_include_filter = self.found_exclude_filter = False
+
+    def process_rows(self, rows, folder_results):
+
+        current_folder = None
+        is_folder_level = config.sections["searches"]["folder_level_filtering"]
+        update_ui = False
+
+        for row in rows:
+            folder_path = row[4]
+
+            if not is_folder_level:
+                if self.check_filter(row):
+                    self.add_row_to_model(row)
+                    update_ui = True
+                continue
+
+            if folder_path != current_folder:
+                self.process_folder_results(folder_results)
+                current_folder = folder_path
+
+            if self.found_exclude_filter:
+                continue
+
+            folder_results.append(row)
+
+            if self.check_filter(row):
+                self.found_include_filter = True
+                update_ui = True
```

Comment:
> I want to avoid setting `self.found_exclude_filter` while filtering

```
else:
     self.found_exclude_filter = True
```

At least having one instance of it here would be better than it being scattered amongst functions.